packages/headless/src/api/knowledge/baseKnowledgeAPISlice.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  BaseQueryFn,
+  createApi,
+  EndpointDefinition,
+  FetchArgs,
+  fetchBaseQuery,
+  FetchBaseQueryError,
+  retry,
+} from '@reduxjs/toolkit/query';
+import {ConfigurationSection} from '../../state/state-sections';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+/**
+ * `dynamicBaseQuery` is passed to the baseQuery of the createApi,
+ * but note that the baseQuery will not be used if a queryFn is provided in the createApi endpoint
+ */
+const dynamicBaseQuery: BaseQueryFn<
+  string | FetchArgs,
+  unknown,
+  FetchBaseQueryError
+> = async (args, api, extraOptions) => {
+  const state = api.getState() as StateNeededByKnowledgeAPI;
+  const updatedArgs = {
+    ...(args as FetchArgs),
+    headers: {
+      ...((args as FetchArgs)?.headers || {}),
+      Authorization: `Bearer ${state.configuration.accessToken}`,
+    },
+  };
+  try {
+    const data = fetchBaseQuery({
+      baseUrl: `${state.configuration.platformUrl}/rest/organizations/${state.configuration.organizationId}`,
+    })(updatedArgs, api, extraOptions);
+    return {data};
+  } catch (error) {
+    return {error: error as FetchBaseQueryError};
+  }
+};
+
+export const baseKnowledgeAPISlice = createApi({
+  reducerPath: 'knowledgeApi',
+  baseQuery: retry(dynamicBaseQuery, {maxRetries: 3}),
+  endpoints: () => ({
```

Comment:
How about : 
```suggestion
  endpoints: () => ({})
```

This seems to be what RTK recommend: https://redux-toolkit.js.org/rtk-query/usage/code-splitting

Reply:
Yup. No need to declare the endpoint firsthand if we use the returning object from the injection. Thanks again.

At first I was pulling on the original definition and got a problem with `(Unfortunately, it cannot modify the types for the original definition.)`